contentpal-api/src/main/java/org/dmfs/android/contentpal/Projection.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.android.contentpal;
+
+import android.support.annotation.NonNull;
+
+
+/**
+ * A column projection for content provider queries.
+ *
+ * @param <T>
+ *         The contract of the view this projection belongs to.
+ *
+ * @author Marten Gajda
+ */
+public interface Projection<T> extends Iterable<String>
+{
+    @NonNull
+    String[] toArray();
```

Comment:
If deduplication of the columns is a critical thing, meaning that SQLite may throw, or duplicate the data in the result, than I think ideally `Projection` type itself would guarantee that, although I am not sure how. If this type is kept, then maybe it could be documented there, that implementations need to ensure that. Both in `iterator` and `toArray()`

Alternative is that `BaseView` does the deduplication, so there is only a single place where `Projection` is applied and duplicates are removed there.

Aren't `iterator()` and `toArray()` sort of redundant? They have to be kept in sync in terms of content.
`extends Iterable` is added for the creating side usage, `toArray()` for the client side usage.
So I think to keep usage simple, but also ensure deduplication, `Projection<T> extends Iterable<String>` could be sufficient, and do the decuplication in `BaseView`.
It could be documented that the consumer of `Projection` need to ensure deduplication.

What do you think, do I miss something?